### PR TITLE
Action Server to execute trees

### PR DESCRIPTION
**Pull request by facontidavide**
Thanks to the hard work of @MarqRazz , we have here a first draft of a "standard" action server to execute trees on demand.

Virtual methods add a number of customization points, that are hopefully general enough to cover 99% of the cases.

### CODE REVIEW

File: btcpp_ros2_samples/src/sample_bt_executor.cpp
Comment (MarqRazz) [inline btcpp_ros2_samples/src/sample_bt_executor.cpp:1]: What do you think about moving this file to the `behaviortree_ros2` package and renaming it to `bt_server` or something along those lines? For the most part people should be able to use this node to run the library and it seems weird to have launch files calling `<node pkg="btcpp_ros2_samples" exec="sample_bt_executor">` when it's a very capable node and not just a sample.

**Review by MarqRazz — COMMENTED** [review]
I have several fixes to get this working and include some basic documentation. Should have them up later today.

**Comment by MarqRazz** [comment]
I added some basic docs and cleaned up a few things so the examples run, let me know what you think. I was traveling most of today but will try and look into the other changes you made later.

File: behaviortree_ros2/src/bt_utils.cpp [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        {
+          RCLCPP_ERROR(kLogger, "Failed to load Plugin from file: %s.", filename.c_str());
+          continue;
+        }
+        RCLCPP_INFO(kLogger, "Loaded ROS Plugin: %s", filename.c_str());
+      }
+      catch(const std::exception& e)
+      {
+        RCLCPP_ERROR(kLogger, "Failed to load ROS Plugin: %s \n %s", filename.c_str(),
+                     e.what());
```
Comment (MarqRazz) [inline behaviortree_ros2/src/bt_utils.cpp:131]: We can drop `ROS` from these prints now that it does both types.

File: behaviortree_ros2/src/tree_execution_server.cpp [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+TreeExecutionServer::Pimpl::Pimpl(const rclcpp::NodeOptions& node_options)
+  : node(std::make_unique<rclcpp::Node>("bt_action_server", node_options))
+{
+  param_listener = std::make_shared<bt_server::ParamListener>(node);
+  params = param_listener->get_params();
+  global_blackboard = BT::Blackboard::create();
+}
+
+TreeExecutionServer::~TreeExecutionServer()
+{}
```
Comment (MarqRazz) [inline behaviortree_ros2/src/tree_execution_server.cpp:71]: Do we need to handle joining and cleaning up the `action_thread` here?

File: behaviortree_ros2/tree_execution_server.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+### std::optional<std::string> onLoopFeedback()
+
+This callback is invoked after `tree.tickOnce` and `onLoopAfterTick`.
+
+If it returns something other than `std::nullopt`, the provided string will be
+sent as feedback to the `rclcpp_action::Client`.
+
+
+
+## ROS Parameters
```
Comment (MarqRazz) [inline behaviortree_ros2/tree_execution_server.md:68]: [generate_parameter_library](https://github.com/PickNikRobotics/generate_parameter_library?tab=readme-ov-file#generate-markdown-documentation) offers the ability to auto generate documentation, which I added in my commit.

File: btcpp_ros2_samples/README.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
+# Sample Behaviors
+
+For documentation on sample behaviors included in this package please see the BehaviorTree.CPP [ROS 2 Integration documentation](https://www.behaviortree.dev/docs/ros2_integration) .
+
+# TreeExecutionServer Documentation and Example
```
Comment (MarqRazz) [inline btcpp_ros2_samples/README.md]: I'm sorry I got out of sync with you on the docs. I will clean up this doc and remove duplicate info.

**Comment by MarqRazz** [comment]
I have a fix for pre-commit and minor changes to the docs if you are okay with me pushing it.

**Comment by facontidavide** [comment]
I was going to merge it... right now! Waiting for CI to turn green. Thanks!

**Comment by facontidavide** [comment]
I will wait for your changes to the docs to be pushed

**Review by MarqRazz — APPROVED** [review]
Thanks for all the help here @facontidavide. My changes were to fix the file permissions (which you fixed) and some minor white space but this looks good as is.

**Pull request merged**